src/partition/repart.c

```diff
@@ -4777,6 +4778,28 @@ static int determine_auto_size(Context *c) {
         return 0;
 }
 
+static int determine_sector_size(const char *node, uint64_t *lbs) {
+        _cleanup_close_ int fd = -1;
+        struct stat st;
+        int r;
+
+        assert(node);
+        assert(lbs);
+
+        fd = open(node, O_RDONLY|O_CLOEXEC);
+        if (fd < 0)
+                return -errno;
+
+        r = fstat(fd, &st);
+        if (r < 0)
+                return -errno;
+
+        if (!S_ISBLK(st.st_mode))
+                return 0;
+
+        ioctl(fd, BLKSSZGET, lbs);
```

Comment:
error checking is missing

Reply:
also use a local variable, and assign to the caller's ref only if the ioctl succeeds, otherwise leave lbs unoutched

Reply:
> also use a local variable, and assign to the caller's ref only if the ioctl succeeds, otherwise leave lbs unoutched

Hmm, is that really necessary though? I don't think (the) ioctl will touch lbs when it fails.

Reply:
it's part of the coding style